heartbeat/ibmcloud-vpc-cr-vip.in

```diff
@@ -0,0 +1,248 @@
+		ext_ip_1, ext_ip_2, api_key)
+        ocf.logger.info(f"start_action: active_ip: {active_ip}")
+    except Exception as e:
+        ocf.logger.error('Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+
+    return ocf.OCF_SUCCESS
+
+def monitor_action(
+    vpc_url="",
+    ext_ip_1="",
+    ext_ip_2="",
+    api_key="",
+):
+    """monitor_action: check if service ip and gateway are responding."""
+
+    ocf.logger.debug("monitor_action:")
+    try:
+        active_ip = ibm_cloud_fail_over.fail_over_cr_vip("GET", vpc_url , ext_ip_1,
+			ext_ip_2, api_key)
+        if ip_address_validator(active_ip):
+            if os_ip(active_ip):
+            	return ocf.OCF_SUCCESS
+        return ocf.OCF_NOT_RUNNING
+    except Exception as e:
+        ocf.logger.error('Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+
+
+def main():
+    """Instantiate the resource agent."""
+
+    agent_description = textwrap.dedent("""\
+        Resource Agent to move a IBM Cloud VIP based on Custom Route
+        from one virtual network interface (VNI) to another.
+        The prerequisites for the use of this resource agent are as follows:
+
+        1. A two-node (VSI or BM) cluster distributed in same Avilability zone or across.
+        2. Enable Instance Metadata enabled on the two nodes
+        3. allow_ip_spoofing enabled on the Virtual network interface
+        3. IBM Cloud API Key or Trused profile:
+
+
+
+    """)
+
+    agent = ocf.Agent(
+        "ibmcloud-vpc-cr-vip",
+        shortdesc="Manages moving move a IBM Cloud VIP based on Custom Route",
+        longdesc=agent_description
+    )
+
+
+    agent.add_parameter(
+        "vpc_url",
+        shortdesc="VPC_URL",
+        longdesc="IBM Cloud Public VPC URL for your region or a VPE URL for IBM Cloud VPC",
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "ext_ip_1",
+        shortdesc="Ip address for first instance.",
+        longdesc="predefined private ip address for first instance.",
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "ext_ip_2",
+        shortdesc="Ip address for secound instance.",
+        longdesc="predefined private ip address for secound instance.",
+        content_type="string",
+        required=True,
+    )
+
+
+    agent.add_parameter(
+        "api_key",
+        shortdesc="API Key or @API_KEY_FILE_PATH",
+        longdesc=(
+            "API Key or @API_KEY_FILE_PATH for IBM Cloud access. "
+            "The API key content or the path of an API key file that is indicated by the @ symbol."
+            "Not needed if trusted profile is used"
+        ),
+        content_type="string",
+        required=False,
+    )
+
+
+    agent.add_action("start", timeout=900, handler=start_action)
+    agent.add_action("stop", timeout=450, handler=stop_action)
```

Comment:
start/stop timeouts can probably be much lower. It's usually 30s or lower for the start-action (60s is a usual high value if the start-action is slow), and 30s or lower for the stop- and monitor-actions.

See example at
https://github.com/ClusterLabs/resource-agents/blob/main/doc/dev-guides/ra-dev-guide.asc#metadata